### PR TITLE
fix(label): only render id on parent calcite-label

### LINKED ISSUE #1106: Bug: [calcite-label] produces duplicate ID elements on the page
### Summary

Setting ID on a calcite-label will cause duplicate ID elements. Setting this as high priority as this causes a11y audits to fail (https://dequeuniversity.com/rules/axe/4.0/duplicate-id?application=axeAPI).

### Actual Behavior

The internal label gets the host's `id` attribute passed through and exposed to the page since the component does not use shadow DOM.

### Expected Behavior

The internal label ID should not be exposed on the page.

### Reproduction Steps

The following codepen illustrates the issue: https://codepen.io/jcfranco-the-scripter/pen/mdEygWa?editors=1010

### PR DESCRIPTION
**Pull request by macandcheese**
Exclude id from attributes spread to child label
Add demo
Adds test

**Related Issue:** Resolves #1106 cc @jcfranco

### CODE REVIEW

**Review by driskull — APPROVED** [review]
👍

**Review by eriklharper — APPROVED** [review]
Looks great

File: src/demos/calcite-label.html [hunk 1]
```diff
@@ -82,6 +82,13 @@ <h3>small wrapping calcite-input</h3>
         <calcite-input></calcite-input>
       </calcite-label>
 
+      <h3>only render id on parent calcite-label</h3>
```
Comment (jcfranco) [inline src/demos/calcite-label.html:85]: The e2e test should cover this, so this is not needed.

**Review by jcfranco — APPROVED** [review]
@macandcheese Thanks for fixing!

File: src/demos/calcite-label.html (hunk 1 above)
Comment (macandcheese) [reply src/demos/calcite-label.html]: Removed 👍

**Pull request merged**